numpy/ma/core.py

```diff
@@ -2968,8 +2968,8 @@ def __array_wrap__(self, obj, context=None):
                 np.copyto(result, fill_value, where=d)
                 # Update the mask
                 if m is nomask:
-                    if d is not nomask:
-                        m = d
+                    if any(d):
```

Comment:
This ends up calling `MaskedArray.any()`; maybe cleaner to just write `d.any()`?

More generally, for the common case of largish arrays with no domain problems, the whole domain now gets checked twice. Might it be an idea to combine with the `copyto` and the whole finding of `fill_value`? I.e.,
```
if d.any():
    try:
      ... get fill_value ...
    np.copyto(result, fill_value, where=d)
    if m is nomask:
        m = True if d.all() else d
    else:
        m = (m | d)
```
(Haven't thought deeply enough to be sure whether or not one could more the `result = result.copy()` inside the if-statement as well.)